Plugin/NodOSVRPlugin.cpp

```diff
@@ -0,0 +1,180 @@
+/** @file
+    @brief Comprehensive example: Implementation of a dummy Hardware Detect
+   Callback that creates a dummy device when it is "detected"
+
+    @date 2014
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+*/
+
+// Copyright 2014 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/PluginKit/PluginKit.h>
+#include <osvr/PluginKit/AnalogInterfaceC.h>
+#include <osvr/PluginKit/ButtonInterfaceC.h>
+#include <osvr/PluginKit/TrackerInterfaceC.h>
+
+// Generated JSON header file
+#include "NodOSVR.h"
+#include "NodPlugin.h"
+
+#pragma comment(lib, "OpenSpatialDll.lib")
+
+// Library/third-party includes
+// - none
+
+// Standard includes
+#include <iostream>
+#define _USE_MATH_DEFINES
+#include <math.h>
+
+// Global device count for path incrementation
+int deviceCount = 0;
+
+// Anonymous namespace to avoid symbol collision
+namespace {
+
+void euler(const float& pitch, const float& roll, const float& yaw, double* mData)
+{
+    const float sinHalfYaw = (float)sin(yaw / 2.0f);
+    const float cosHalfYaw = (float)cos(yaw / 2.0f);
+    const float sinHalfPitch = (float)sin(pitch / 2.0f);
+    const float cosHalfPitch = (float)cos(pitch / 2.0f);
+    const float sinHalfRoll = (float)sin(roll / 2.0f);
+    const float cosHalfRoll = (float)cos(roll / 2.0f);
+
+    mData[0] = cosHalfRoll * cosHalfPitch * cosHalfYaw
```

Comment:
Please add some comments on what each formula calculates or explicitly name contents of quaternion because right now it hard to say which value is W, X, Y or Z.